lib/media/track/localvideotrack.js

```diff
@@ -27,7 +30,32 @@ class LocalVideoTrack extends LocalMediaVideoTrack {
    * @param {LocalTrackOptions} [options] - {@link LocalTrack} options
    */
   constructor(mediaStreamTrack, options) {
+    options = Object.assign({
+      workaroundSilentLocalVideo: guessBrowser() === 'safari'
+        && typeof document !== 'undefined'
+        && typeof document.createElement === 'function'
+    }, options);
+
     super(mediaStreamTrack, options);
+
+    Object.defineProperties(this, {
+      _workaroundSilentLocalVideo: {
+        value: options.workaroundSilentLocalVideo
+          ? workaroundSilentLocalVideo
```

Comment:
I would keep it boolean than keeping function value. It's not writable, so don't gain much by keeping function value in this property.  By  keeping function value gives access to it to the api callers - which is not needed.

Reply:
if you want to keep it function then maybe assign it a default value that does nothing...that way you can get rid of `if (this._workaroundSilentLocalVideo)` checks.

```
    Object.defineProperties(this, {
      _workaroundSilentLocalVideo: {
        value: options.workaroundSilentLocalVideo
          ? workaroundSilentLocalVideo
          : () => null
      },
      _workaroundSilentLocalVideoCleanup: {
        value: null,
        writable: true
      }
    });

this._workaroundSilentLocalVideoCleanup = this._workaroundSilentLocalVideo(this, document);

```

Reply:
@makarandp0 I had initially written it that way, but it would not be clear to the reader if that function is called without conditions. So, I think I will keep it as it is for now.